lib/spack/spack/variant.py

```diff
@@ -349,6 +348,9 @@ def satisfies(self, other):
         # (`foo=bar` will never satisfy `baz=bar`)
         return other.name == self.name
 
+    def intersects(self, other):
+        return self.compatible(other)
+
     @implicit_variant_conversion
     def compatible(self, other):
         """Returns True if self and other are compatible, False otherwise.
```

Comment:
Minor nitpick: can you make `compatible` call `intersects` and have the implementation in `intersects` for consistency? The docstring probably should also say "returns true if there are concrete specs matching both lhs and rhs"

Reply:
Done in https://github.com/spack/spack/pull/35681/commits/bebaf82a9fee74cb45e746ef67d27e725b2ede76